Assets/BossRoom/Scripts/Shared/ScriptableObjects/GameEventListener.cs

```diff
@@ -0,0 +1,38 @@
+using System;
+using UnityEngine;
+using UnityEngine.Assertions;
+using UnityEngine.Events;
+
+namespace BossRoom
+{
+    /// <summary>
+    /// This class is designed to work hand in hand with the GameEvent class, which is a ScriptableObject container
+    /// for an event. GameEventListener declares a GameEvent, which is to be assigned within the editor. When the
+    /// GameEvent's Raise() method is fired, this class will invoke a UnityEvent.
+    /// </summary>
+    public class GameEventListener : MonoBehaviour
```

Comment:
I would also prefer having an Interface vs an MB Class so that users can implement that and still subscribe to events even if they don't inherit from a GameEventListener.

Reply:
100% agree on this. I've added that and removed the previous MonoBehaviour listener, added the interface, and a better named component for raising UnityEvents.